server/src/main/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResource.java

```diff
@@ -38,11 +50,65 @@
 
   private final DSLContext jooq;
   private final RowHmacGenerator rowHmacGenerator;
+  private final AuditLog auditLog;
 
   @Inject
-  public BackfillRowHmacResource(DSLContext jooq, RowHmacGenerator rowHmacGenerator) {
+  public BackfillRowHmacResource(
+      DSLContext jooq,
+      RowHmacGenerator rowHmacGenerator,
+      AuditLog auditLog) {
     this.jooq = jooq;
     this.rowHmacGenerator = rowHmacGenerator;
+    this.auditLog = auditLog;
+  }
+
+  /**
+   * Backfill row_hmac for a specific secret.
+   */
+  @Timed @ExceptionMetered
+  @Path("secret/name/{secretName}")
+  @POST
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  public void backfillSecretRowHmacByName(
+      @Auth AutomationClient automationClient,
+      @PathParam("secretName") String secretName,
+      @QueryParam("force") boolean force) {
+    Optional<SecretsRecord> maybeRow = jooq.selectFrom(SECRETS)
+        .where(SECRETS.NAME.eq(secretName))
+        .fetchOptional();
+
+    if (!maybeRow.isPresent()) {
+      throw new NotFoundException();
+    }
+
+    SecretsRecord row = maybeRow.get();
+    String oldHmac = row.getRowHmac();
+
+    if (oldHmac != null && !force) {
+      return;
+    }
+
+    String newHmac = rowHmacGenerator.computeRowHmac(
```

Comment:
would probably be a good time to standardize computing the row hmac instead of having arbitrary parameters and repeating this everywhere

Reply:
I looked into this, but it's a slightly bigger change than I was expecting. Calculating the HMAC on persistence is done against one model, but calculating the HMAC on load is done against the low-level row model, so we'd have to change around the guts a bit if we want to have, say, a single utility method on the RowHmacGenerator class. OTOH I was also thinking that we could add methods to the model classes themselves. But yeah, a follow-up change is to add the owner to the row HMAC so I was planning on incorporating the refactoring work as part of that work.